docs/writing-easyconfig-files.md

```diff
@@ -940,6 +940,73 @@ Example:
+
+1. Copy the easyconfig and change the toolchain version
+1. Usually, especially when updating to the latest toolchain,
+   you should search for the most recent version of the software too and use that.
+1. For each listed dependency find the easyconfig for the new toolchain version.
+   Here you need to take [toolchain hierarchy](common-toolchains.md#toolchains_diagram) into account.
+   I.e. a dependency of an easyconfig for the `foss` toolchain might use the `GCC` or `GCCcore` toolchain.
+   Check the toolchain easyconfig file (e.g. `foss-2023b.eb`) to find the versions of the sub-toolchain.
+   For very recent toolchains you can also check the `develop` branch of the [easyconfigs git repository](https://github.com/easybuilders/easybuild-easyconfigs) and open PRs.
+1. If an easyconfig for the listed dependency with the new toolchain already exists use its version in your new easyconfig.
+   Otherwise, you need to create a new easyconfig for the listed dependency with the new toolchain version (based on that of another toolchain version).
+   The steps are the same and also need to be repeated for each dependency of this dependency.
+1. Finally, build and test your easyconfig and consider [contributing](contributing.md#contributing-easyconfig-files--contributing_easyconfigs) it.
+
+Especially when creating easyconfigs for dependencies it is a good idea to put all of them into a new folder.
+To allow EasyBuild to find these easyconfigs in your folder, pass its path via `eb --robot <path> ...`.
+
+This manual process can be partially automated:
+
+1. Create a new folder for the new easyconfig and its dependencies.
+1. Run `eb <current_easyconfig.eb> --try-toolchain-version=<new-version> --try-update-deps --experimental --copy-ec <folder>`.
+   This will update the toolchain version, search for the versions of dependencies in the new toolchain and use them if available.
+   If a dependency with the new toolchain is not available this will create a new easyconfig for this toolchain
+   based on an existing one for another toolchain (version).
+   All newly create easyconfigs will be copied to the specified folder.
+1. Optionally, but recommended, update the version of the software for each created easyconfig to the latest available one.
+1. **Important**: Verify the newly created easyconfigs by comparing each of them against the existing one for that software.
+   Only the version of the software, toolchain and dependencies should have been changed.
+   EasyBuild often replaces local variables or templates like `name` or `%(version)s` by their value when updating easyconfigs.
+   This is undesirable as the values might become out of sync with what they are supposed to be, e.g. after changing the version variable.
+   It might be easier to use those easyconfigs just for getting the versions of each dependency required
+   and still copy & update the easyconfigs from existing ones manually.
```

Comment:
```suggestion
   Only the version of the software, toolchain and dependencies should have been changed. However, this approach
   can lead to unwanted changes to formatting as well as the replacement local variables or templates (such as `name`,
   `version` or `%(version)s`) by their _value_ when updating easyconfigs.
   This is undesirable as the values might become out of sync with what they are supposed to be, e.g. when we update
   the `version` variable the old _value_ may still exist elsewhere in the easyconfig.
   Depending on the situation, it might be easier to use the newly created easyconfigs just for getting the versions of
   each dependency required and still copy & update the easyconfigs from existing ones manually.
```